tests/std/tests/GH_003965_robust_against_adl_any/test.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <any>
+#include <utility>
+
+struct incomplete;
+
+template <class T>
+struct wrapper {
+    T t;
+};
+
+struct nontrivial {
+    nontrivial() {}
+    ~nontrivial() {}
+};
+
+int main() {
+    using foo = wrapper<incomplete>*;
+    struct bar {
+        foo p{};
+        nontrivial q{};
+    };
+    struct baz {
+        foo p[20]{};
+    };
+
+    std::any a{foo{}}, b(bar{}), c{baz{}};
+    std::any *ap = &a, *bp = &b, *cp = &c;
+    a = *ap, b = *bp, c = *cp;

```

Comment:
if the pointers are to fool the compiler to suppress some waring, I'd consider actual suppression instead to make it more obvious what you do here.

Reply:
I've suppressed the warnings in this file. As to the one in `P0220R1_any`, as there is only one such usage I think it's ok to keep it unchanged.

Reply:
No change requested: there can be a concentrated test for self move (and/or copy) assignment in the future, and we may close #1485 at that time.